protos/backend_service.proto

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+syntax = "proto3";
+
+package durabletask.protos.backend.v1;
+
+option csharp_namespace = "Microsoft.DurableTask.Protobuf";
+option java_package = "com.microsoft.durabletask.implementation.protobuf";
+option go_package = "/internal/protos";
+
+import "orchestrator_service.proto";
+
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/empty.proto";
+
+// gRPC service used by Durable Task Framework (DTFx) backend implementations.
+// The RPCs in this service are used by DTFx clients to manage orchestrations.
+service ClientBackendService {
+    // Creates a new orchestration instance.
+    rpc CreateOrchestrationInstance (CreateInstanceRequest) returns (CreateInstanceResponse);
+
+    // Sends an event to an orchestration instance. This RPC is used for raising external events to orchestrations
+    // and for sending orchestration lifecycle events, such as terminate, suspend, resume, etc.
+    rpc AddOrchestrationEvent (AddOrchestrationEventRequest) returns (google.protobuf.Empty);
+
+    // Returns metadata about an orchestration instance.
+    rpc GetOrchestrationMetadata (GetInstanceRequest) returns (GetInstanceResponse);
+
+    // Waits for an orchestration to reach a terminal state and then returns metadata for that orchestration.
+    rpc WaitForOrchestrationCompletion(GetInstanceRequest) returns (GetInstanceResponse);
+
+    // Purges the state of one or more orchestration instances.
+    rpc PurgeOrchestrationState (PurgeInstancesRequest) returns (PurgeInstancesResponse);
+
+    // TODO: Additional operations, such as listing orchestrations
+}
+
+// gRPC service used by Durable Task Framework (DTFx) backend implementations.
+// The RPCs in this service are used by the DTFx worker to process orchestration and activity work items.
+service WorkerBackendService {
+    // Starts a bi-directional stream for receiving work items and sending health pings.
+    rpc GetWorkItems(stream GetWorkItemsRequest) returns (stream WorkItem);
```

Comment:
Recommend not using `WorkItem` as response, but a message that includes `WorkItem` as a `oneof`. This way we can potentially add more message types in the future if needed

Reply:
The `WorkItem` field itself is already a `oneof`, so we can create new types of work items over time if needed (which I've done already in the next commit - the health ping work item response).